ygen/gogen.go

```diff
@@ -218,6 +218,25 @@ type generatedGoListMethod struct {
 	Receiver  string          // Receiver is the name of the parent struct of the list, which is the receiver for the generated method.
 }
 
+// generatedGoKeyHelper contains the fields required for generating a method
+// associated with a struct that is within a list in the YANG schema.
+type generatedGoKeyHelper struct {
+	// Receiver is the name of the type which acts as a receiver for a generated method.
+	Receiver string
+	// Keys specifies the keys of the list, as a map from YANG to Go identifier.
+	Keys []*yangFieldMap
+}
+
+// yangFieldMap maps a YANG identifier to its Go identifier.
+type yangFieldMap struct {
+	// YANGName is the field's name in the YANG schema.
+	YANGName string
+	// GoName is the field's name in the Go struct.
+	GoName string
+	// IsPtr indicates that the key field is a pointer.
```

Comment:
This seems a bit strange. Is it not possible to infer this from someplace?

Reply:
It isn't, since this struct is only used as input to a template, so the template has no real logic to be able to know whether the strings that it received is a pointer or not. The ideal here is that there is zero logic other than very simple `if`/`else`/`range` in the template as they are terrible to maintain.